### PR TITLE
[EWL-3093] : uses gulp to inject js needed to create a scss tab panel

### PR DESCRIPTION
**Pull request by rpama**
## Ticket(s)
**Jira Ticket**

- [EWL-3093: Get CSS in Tab](https://issues.ama-assn.org/browse/EWL-3093)

## Description

Grossly misuse Gulp to inject some panel code into patternlab source files instead of having to fork from their repos or patch.

## To Test

- [ ] Navigate to `styleguide/` and run `npm install`
- [ ] run `gulp serve`
- [ ] Click upper right Settings gear > Show Pattern Info
- [ ] See that scss panel exists
- [ ] Go to a pattern with scss (so not the intro) and see that the scss is available in the pattern

## Relevant Screenshots/GIFs

screen shot 2017-06-16 at 10 53 08 am

## Remaining Tasks

Should probably add a try/catch in the php so the log doesn't show warning when no scss file exists.

## Additional Notes

We should decide if we like this approach and if we want to try to add other panels, show css instead of scss, and/or if we require styling to the markup (not sure yet how easily that might be accomplished).

### CODE REVIEW

**Comment by froboy** [comment]
Poking around in the source, `styleguide/.publish/styleguide/js/patternlab-viewer.js` line 765 seems to be where the syntax coloring comes in and http://prismjs.com/ has some good documentation on how it works.

File: styleguide/gulpfile.js [hunk 1]
```diff
@@ -107,9 +108,9 @@ gulp.task('minifyIcons', function() {
 });
 
 // Based on https://github.com/filamentgroup/gulpicon#usage
-var iconFiles = glob.sync("source/assets/icons/svg/*.svg");
-var iconConfig = require("./source/assets/icons/config.js");
-iconConfig.dest = "public/assets/icons/";
+var iconFiles = glob.sync('source/assets/icons/svg/*.svg');
```
Comment (froboy) [inline styleguide/gulpfile.js]: Still a few stray quotes

Comment (rpama) [inline styleguide/gulpfile.js]: These lines were out of sync with the single-quote standard we use everywhere else which is why I left this correction.

**Comment by Brian-Clement** [comment]
@rpama @froboy I've not been able to play with this yet with my `gulp` currently broken, but what are your thoughts on this: https://github.com/cweagans/composer-patches

It looks like you can just create a patch for the `vendor/pattern-lab/core/src/PatternLab/Builder.php` file (or any other), commit it to the repo here and update `composer.json` (and add `composer.patches.json`) and have it apply for you.  

Not sure if this is overkill or not, but this looked like a slick package.

**Comment by rpama** [comment]
@froboy Despite the fact that they use prismjs, I did some preliminary tinkering with this and did not have any success getting code highlighting to work. It could have to do with how the styleguidekit pulls prism in, but it's not looking like this would be an easy fix. It think we should decide if the code is of any use to use without that and make a separate ticket if code highlighting is a nice-to-have.

**Comment by rpama** [comment]
Have created a new PR with patches at https://github.com/AmericanMedicalAssociation/AMA-style-guide/pull/149

**Comment by froboy** [comment]
#149 is merged, so closing this accordingly.

**Pull request closed without merging**